exercises/variable-length-quantity/src/test/java/VariableLengthQuantityTest.java

```diff
@@ -0,0 +1,248 @@
+        List<String> expected = Arrays.asList("0x8f", "0xff", "0xff", "0xff", "0x7f");
+        List<Long> numbers = Arrays.asList(0xffffffffL);
+
+        assertEquals(expected, variableLengthQuantity.encode(numbers));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testTwoSingleByteValues() {
+        List<String> expected = Arrays.asList("0x40", "0x7f");
+        List<Long> numbers = Arrays.asList(0x40L, 0x7fL);
+
+        assertEquals(expected, variableLengthQuantity.encode(numbers));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testTwoMultiByteValues() {
+        List<String> expected = Arrays.asList("0x81", "0x80", "0x0", "0xc8", "0xe8", "0x56");
+        List<Long> numbers = Arrays.asList(0x4000L, 0x123456L);
+
+        assertEquals(expected, variableLengthQuantity.encode(numbers));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testManyMultiByteValues() {
+        List<String> expected = Arrays.asList("0xc0", "0x0", "0xc8", "0xe8", "0x56", "0xff", "0xff", "0xff", "0x7f", "0x0", "0xff", "0x7f", "0x81", "0x80", "0x0");
+        List<Long> numbers = Arrays.asList(0x2000L, 0x123456L, 0xfffffffL, 0x0L, 0x3fffL, 0x4000L);
+
+        assertEquals(expected, variableLengthQuantity.encode(numbers));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testDecodeOneByte() {
+        List<String> expected = Arrays.asList("0x7f");
+        List<Long> bytes = Arrays.asList(0x7fL);
+
+        assertEquals(expected, variableLengthQuantity.decode(bytes));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testDecodeTwoBytes() {
+        List<String> expected = Arrays.asList("0x2000");
+        List<Long> bytes = Arrays.asList(0xc0L, 0x0L);
+
+        assertEquals(expected, variableLengthQuantity.decode(bytes));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testDecodeThreeBytes() {
+        List<String> expected = Arrays.asList("0x1fffff");
+        List<Long> bytes = Arrays.asList(0xffL, 0xffL, 0x7fL);
+
+        assertEquals(expected, variableLengthQuantity.decode(bytes));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testDecodeFourBytes() {
+        List<String> expected = Arrays.asList("0x200000");
+        List<Long> bytes = Arrays.asList(0x81L, 0x80L, 0x80L, 0x0L);
+
+        assertEquals(expected, variableLengthQuantity.decode(bytes));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testDecodeMaximum32BitInteger() {
+        List<String> expected = Arrays.asList("0xffffffff");
+        List<Long> bytes = Arrays.asList(0x8fL, 0xffL, 0xffL, 0xffL, 0x7fL);
+
+        assertEquals(expected, variableLengthQuantity.decode(bytes));
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void testCannotDecodeIncompleteSequence() {
+        List<Long> bytes = Arrays.asList(0xffL);
+
+        assertEquals(null, variableLengthQuantity.decode(bytes));
```

Comment:
Since this is essentially an error case I would maybe expect it to throw an exception instead of returning `null`?

Reply:
This is a really good point to raise actually - in problem specifications, most of the time when the canonical data expects `null`, this is just an indicator to language tracks that we should handle this error case in whichever way is most natural to the language.

Reply:
That's a good point; I'll go ahead and change that. Maybe something like this?
```java
throw new IllegalArgumentException("Invalid variable-length quantity encoding");
```

Reply:
Sounds good! 🙂